frontends/article/article-page.tsx

```diff
@@ -5,67 +5,60 @@ import Image from 'next/image';
 import { FC, useEffect, useMemo, useState } from 'react';
 import { useAppDispatch } from 'store';
 import { useLazyCheckIfLikedDislikedQuery, useLikeDislikeMutation } from 'store/apis';
-import { openLoginModal, toggleCommentsSidebar } from 'store/states';
+import { openLoginModal } from 'store/states';
 import { addUriToImageBlocks, toDateString } from 'utils';
-import { UPPER_BLUE_COLOR } from 'variables';
-import { ICONS } from 'variables/icons';
+import { ICONS, UPPER_BLUE_COLOR } from 'variables';
 
 import styles from './article.module.scss';
 import { IArticleProps } from './article.types';
-import { ArticleActions } from './components';
-
-const LikeIcon = ICONS.like;
-const DislikeIcon = ICONS.dislike;
-const CommentIcon = ICONS.comment;
+import { ArticleActionIcons, ArticleActions } from './components';
 
 const toUzbDateString = (date: Date | string): string => toDateString(date, { month: 'short' });
+const LikeIcon = ICONS.like;
 
-export const Article: FC<IArticleProps> = (props) => {
-  const { viewCount, publishedDate, updatedDate, blocks, id, likeCount, dislikeCount } = props;
-  const [editorInstance, setEditorInstance] = useState<EditorJS | null>(null);
+export const Article: FC<IArticleProps> = (article) => {
+  const {
+    id,
+    viewCount = 0,
+    publishedDate,
+    updatedDate,
+    blocks = [],
+    likeCount = 0,
+    dislikeCount = 0,
+  } = article || {};
   const [likeDislikeCount, setLikeDislikeCount] = useState<number>(likeCount - dislikeCount);
+  const [editorInstance, setEditorInstance] = useState<EditorJS | null>(null);
+  const [isSharePopupOpen, setIsSharePopupOpen] = useState<boolean>(false);
+  const [likeDislikeArticle, likeDislikeRes] = useLikeDislikeMutation();
   const { isAuthenticated } = useAuth();
-  const { themeColors } = useTheme();
   const dispatch = useAppDispatch();
-  const [likeDislikeArticle, likeDislikeRes] = useLikeDislikeMutation();
   const [checkIfLikedDislikedQuery, { data: isLikedOrDisliked }] =
     useLazyCheckIfLikedDislikedQuery();
+  const { themeColors } = useTheme();
 
   const likeDislike = (value: -1 | 1): void => {
```

Comment:
all action icon related logic should be inside article-action-icons.tsx component, just get data as props and do the logic inside it